src/vaadin-radio-group.html

```diff
@@ -203,7 +203,10 @@
               if (this.disabled) {
                 button.disabled = true;
               }
-              if (button.checked) {
+              if (this.value !== undefined && this.value == button.value) {
```

Comment:
There's a slightly behavior change here:

The radio button group goes on reverse order of the buttons list to set the last checked button as the value and, when it finds a checked button, then it un-checks the other buttons. But, with this change, if there's another button with the same value as the checked one (which is, for instance, the case when no `value` is defined for the button - defaults to `on`), then this condition will be `true` (`this.value == button.value`) and it will check this button instead. Although the group value remains the same.

Reply:
true. I didn't realize that the buttons don't need to have unique values nor that they don't actually need a value at all.

Reply:
Also it doesn't work if it considers the button checked state as now if you click B then C it changes to C, but then you can't come back to B as C will have the checked flag and will then set the value to C also when selecting. Nor can you go to A from B as B will be checked and take the selection and update the value due to the `else if (button.checked)` before we test for A